### DISCUSSION TITLE
Make Number of laps editable

### DISCUSSION THREAD

**Discussion by Volker-F3F** [gerhardf1/F3F-Timer#5] [Ideas]
Hello Gerhard,
Great tool which I use for F3F practice. To practice  a round as flown during competition (10-laps) the tool fits me perfectly.

When I am practicing and try new setups or different flying styles, from my experience flying 10-laps in one go are not enough to create a reliable average - 1-lap wrong and the average is too much influenced. Typically I fly 30-50 laps and take the flown time with the stopwatch integrated in my Graupner MC28 radio.
Sometimes, when there was a critical situation during such a 30-50 lap round, I miss what is the next lap number and have to repeat the flight. It will be great if I could adjust the number of laps at your App and it works with counting down as currently.
I know at least one other guy who appreciates such feature, too.

Sending kind regards from Graz
Volker

**Comment by gerhardf1** [comment]
Hello Volker,
good idea, but I would change the counting direction:
* Counting the  laps upwards at every base crossing until new button "Last" would be pressed.
* Timer is started after button "Start" was pressed and plane entered the course at what ever base.
* Timer is stopped after button "Last" was pressed and the next base crossing.
* App shows total and average time.

This method would have the advantage that you can fly as many laps as you like and stop measuring at any time.

Would this be fine for you?

regards
Gerhard

**Reply by Volker-F3F** [reply]
Dear Gerhard,
Thank you for your fast reply.

From your reply I understand that you will create a new mode and new button "Last". For me a new mode is not necessary. Keep everything as it is. The current App counts upwards (1...2...3... ...Last) and announces the last turn. All this is perfect for me. I do not want to press a button "Last" on the smartphone while I am in a round ... this can be very dangerous.
I do not know if you fly F3F. There are slopes where you fly 1-2m away from the edge of the slope and it is not possible to take the eyes away from the plane for even half a second (at good wind and thermal condition the plane reaches up to 160km/h at the crossover between turns).

The number of laps I desire to fly must be entered before I launch the plane and then being counted upwards until the desired number of laps was reached and "Last" announced ... identical to current 10-lap mode, just being able to enter the number of laps.
If it is necessary to create a new mode where, when selecting the mode, I have to enter each time the number of laps, that will be OK.

If you add this feature and after some testing at the slope, I will promote your App. Most people I know use the F3F-Timer from Darjes and I prefer your App.
Kind regards From Graz
Volker

**Reply by gerhardf1** [reply]
Hello,
I understand your arguments.

**Reply by Volker-F3F** [reply]
Hello Gerhard,
Thank you for providing a Beta that fast.
My next Training is planned for 13./14.07.24 and I will give you my feedback.

Have a great weekend
Sending kind regards from Graz
Volker

**Reply by Volker-F3F** [reply]
Hello Gerhard,
Yesterday I took the PiCams with me to the model plane airfield and tested them with the Electro Version of the Freestyler6. Everything worked fine. Thank you for your fast implementation!
Sending best regards from Graz, Volker

**Reply by gerhardf1** [reply]
Hello Volker,
good to hear that the new version worked so far.

I released V.011 where the numer of laps are now selectable.

regards
Gerhard
